src/cmd/ksh93/sh/parse.c

```diff
@@ -1439,9 +1439,10 @@ static Shnode_t *simple(Lex_t *lexp,int flag, struct ionod *io)
 		{
 			if(!(argp->argflag&ARG_RAW))
 				argno = -1;
-			if(argno>=0 && argno++==cmdarg && !(flag&SH_ARRAY) && *argp->argval!='/')
+			if(argno>=0 && argno++==cmdarg && !(flag&SH_ARRAY)
+			&& !(sh_isoption(SH_RESTRICTED) && strchr(argp->argval,'/')))
```

Comment:
I know this is minor, but is there a reason why my patch was changed to use `strchr` is used instead of `*argp->argval=='/'`? I'd think that the latter would be a tiny bit faster as far as string comparison is concerned.

Reply:
It is to avoid pointlessly searching the built-ins tree for a non-absolute path like `some/dir/command`, so it seems to me that should save more cycles than it costs, though I could be wrong. In any case the `strchr` is only executed if the restricted option is on.

Reply:
That makes sense.